transaction/core/src/account_keys.rs

```diff
@@ -261,14 +319,20 @@ impl AccountKey {
             RistrettoPublic::from(&subaddress_spend_private)
         };
 
-        match self.fog_url() {
-            None => PublicAddress::new(&subaddress_spend_public, &subaddress_view_public),
-            Some(fog_url) => PublicAddress::new_with_fog(
-                &subaddress_spend_public,
-                &subaddress_view_public,
-                fog_url,
-            ),
+        let mut result = PublicAddress {
+            view_public_key: subaddress_view_public,
+            spend_public_key: subaddress_spend_public,
+            fog_url: self.fog_url.clone(),
+            fog_authority_sig: Default::default(),
+            version: self.version,
+        };
+
+        if !self.fog_url.is_empty() {
```

Comment:
Style-wise, this could use the Option accessor to hide the details of an empty string representing the absense of a Fog url.

Reply:
I think I see what you are saying but I don't know who we are hiding the details from, this is all implementation details of the same object

Reply:
Not in terms of hiding from something, but just in terms of having the logic go through the same code path w.r.t. whether the AccountKey is considered to use Fog or not, so that it's less likely to crop up as subtle bugs later if that logic were to change (e.g. empty string no longer signifies that the AccountKey doesn't use Fog).

I agree that since it's an implementation detail of the same object, it's less of an issue. Feel free to ignore this PR comment though if you prefer how it is.